### DISCUSSION TITLE
reference to a model with a multiple fields based ID

### DISCUSSION THREAD

**Discussion by yuedai0531** [prisma/prisma#4395] [Q&A] [answered]
Hi!

**tl;dr
How to add a relation to a model that has an ID that is defined but multiple fields?**

I have a Conversation model and a Message model. messages have a many-to-one relation to a conversation.
My Conversation model's ID is defined with multiple fields, done following [this part of the docs](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-an-id-field)

```prisma
model Conversation {
  created   DateTime @default(now())
  lawyer    Lawyer   @relation(fields: [lawyerId], references: [id])
  lawyerId  String
  human     Human    @relation(fields: [humanId], references: [id])
  humanId   String
  @@id([humanId, lawyerId])
}

model Message {
  id                String   @default(uuid()) @id
  conversation      Conversation @relation(fields:[conversationId], references: [id]) // <--------
  conversationId    String
  content           String
  created           DateTime @default(now())
  senderUserType    UserType
  senderId          String
}
```

when I run `npx prisma migrate save --experimental`
with either of the following replacing the line I point to
`conversation      Conversation @relation(fields:[conversationId], references: [id])`
`conversation      Conversation @relation(fields:[conversationId], references: [humanId_lawyerId])`

I get these respective errors
```
Environment variables loaded from prisma/.env
 ERROR  Oops, an unexpected error occured!
error: Error validating: The argument `references` must refer only to existing fields in the related model `Conversation`. The following fields do not exist in the related model: id
```
```
Environment variables loaded from prisma/.env
error: Error validating: The argument `references` must refer only to existing fields in the related model `Conversation`. The following fields do not exist in the related model: humanId_lawyerId
```

Thanks!

**Comment by GeoffreyDick** [comment, marked as answer]
Hey I'm pretty new to Prisma but I think the issue is how you've defined the relationship between the Message and Conversation models. Going off the [One-to-Many documentation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#one-to-many-relations) (specifically where it mentions "multi-field IDs"), try the following
```prisma
model Conversation {
  created   DateTime @default(now())
  lawyer    Lawyer   @relation(fields: [lawyerId], references: [id])
  lawyerId  String
  human     Human    @relation(fields: [humanId], references: [id])
  humanId   String
  messages  Message[]
  @@id([humanId, lawyerId])
}

model Message {
  id                 String   @default(uuid()) @id
  conversation       Conversation @relation(fields: [conversationHuman, conversationLawyer], references: [humanId, lawyerId])
  conversationHuman  String
  conversationLawyer String
  content            String
  created            DateTime @default(now())
  senderUserType     UserType
  senderId           String
}
```

I didn't test that answer so let me know if it works or not

**Reply by yuedai0531** [reply]
works! thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
